cmd/configschema/configschema/resolver_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configschema
+
+import (
+	"go/build"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func TestExternalPkgDir(t *testing.T) {
+	pkgPath, _ := externalPackageDir(testDR(), "grpc-ecosystem/grpc-gateway")
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		goPath = build.Default.GOPATH
+	}
+	assert.Equal(t, goPath+"/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.16.0", pkgPath)
```

Comment:
This test would need to be updated every time this dep changes, so a regex could be preferable.

Reply:
A go.mod file in a `testdata` directory makes sense to me and that way you'd be able to have hardcoded expected values like you initially did.  It would need to be ignored by dependabot though: https://github.com/open-telemetry/opentelemetry-collector/blob/main/Makefile#L20

Reply:
I don't think this request warrants blocking the PR as is though and would be good to do in the future.